File: sample-domain/npm/src/aggregates/sample-unit-of-work.ts
import { IUnitOfWork, IUnitOfWorkFactory } from "../abstractions/unit-of-work";
import { IParentRepository } from "./parent/parent-repository";

export interface ISampleUnitOfWork extends IUnitOfWork {
    parents: IParentRepository;
}

export interface ISampleUnitOfWorkFactory extends IUnitOfWorkFactory<ISampleUnitOfWork> { }

File: sample-domain/dotnet/src/aggregates/sample-unit-of-work.cs
using Sample.Domain.Abstractions;
using Sample.Domain.Aggregates.Parent;

namespace Sample.Domain.Aggregates
{
    public interface ISampleUnitOfWork : IUnitOfWork
    {
        IParentRepository Parents { get; }
    }

    public interface ISampleUnitOfWorkFactory : IUnitOfWorkFactory<ISampleUnitOfWork> { }
}